### DISCUSSION TITLE
_all_docs endpoint returns deleted docs when keys are specified in the body

### DISCUSSION THREAD

**Discussion by ykodgule** [apache/couchdb#2318] [Ideas]
## Description
While using _all_docs end point to do a search for the records with specific keys like below:

curl -X POST 'http://X.X.X.X:5984/analysis_db/_all_docs' -u <USER>:<PASSWORD> -H 'Content-Type:application/json' -d '{"keys":["TOP_RISK_2","TOP_RISK_3"]}'

It returns a response as:
{"total_rows":192,"offset":null,"rows":[
{"id":"TOP_RISK_2","key":"TOP_RISK_2","value":{"rev":"54-fa0e4904d88c5d74d4716ac51b311756","deleted":true}},
{"id":"TOP_RISK_3","key":"TOP_RISK_3","value":{"rev":"38-baf385f4bf0cca586a1f8539f35bd2a3","deleted":true}}
]}

However, this is returning deleted documents as well.

## Steps to Reproduce
1) Create Docs in a non-partitioned DB. Assume the _ids are: TOP_RISK_2 and TOP_RISK_3
2) Delete the docs.
3) Now execute the curl command for _all_docs as detailed above.
This returns deleted documents in the response. 

## Expected Behaviour
_all_docs endpoint should return documents which are not deleted or 
should provide a flag that will override the default behavior to return the documents which are not deleted.

## Your Environment
Couchdb3 downloaded from ibmcom/couchdb3 repo

CURL Command:
curl -X POST 'http://X.X.X.X:5984/analysis_db/_all_docs' -u <USER>:<PASSWORD> -H 'Content-Type:application/json' -d '{"keys":["TOP_RISK_2","TOP_RISK_3"]}'

Response:
{"total_rows":192,"offset":null,"rows":[
{"id":"TOP_RISK_2","key":"TOP_RISK_2","value":{"rev":"54-fa0e4904d88c5d74d4716ac51b311756","deleted":true}},
{"id":"TOP_RISK_3","key":"TOP_RISK_3","value":{"rev":"38-baf385f4bf0cca586a1f8539f35bd2a3","deleted":true}}
]}
* CouchDB Version used: CouchDB3
* Browser name and version: Using CuRL
* Operating System and version: RHEL 7.5

**Comment by davisp** [comment]
For comparison, if the document never existed the response includes rows of the form:

`{"key": "missing_key", "error": "not_found"}`

Are you wanting a feature where you can ask to not have rows returned at all? Or wanting to not have to deal with the difference between `"error": "not_found"` and `"deleted": true`?
